### DISCUSSION TITLE
webot simulates sphere in sphere contact forces

### DISCUSSION THREAD

**Discussion by Rozlj** [cyberbotics/webots#4272] [Q&A]
Hi, guys! 
I am using webots to simulate a differential-driven trolley to rotate and forward inside a sphere. It seems that using sphere boundingobject provided by the webots can't generate contact detection on the inner wall of the sphere.  Can anyone give some advice to realize the contace force and friction between the wheels and innner wall of the sphere? THANKS! 

I have tried import a sphere that represents boundingobject using Indexfaceset from a third-party software, there is friction i wanted. However, webots gives the warning >> Your world may be too complex. If this problem persists, try simplifying your bounding object(s).

**Comment by omichel** [comment]
You shouldn't import meshes for bounding objects, but rather use built-in primitives, such as [Sphere](https://cyberbotics.com/doc/reference/sphere), [Box](https://cyberbotics.com/doc/reference/box), [Capsule](https://cyberbotics.com/doc/reference/capsule), etc. This is much more efficient for the collision detection engine.

**Reply by Rozlj** [reply]
Thanks, omichel. 
I have tried built-in sphere. I think that the outer surface can detect its collision with other stuff, for example, running on  a ground. However, if we have a car moving inside the sphere, like a sphero robot, there is no collision detection between the wheels and inner surface of the sphere, which is what I need. 
Any further suggestion would be appreciated.

**Reply by omichel** [reply]
Yes you are right. I overlooked your question and I was wrong with my initial suggestion as the sphere primitive is only an outer sphere, not an inner sphere. Unfortunately, I have no better idea...

**Reply by Rozlj** [reply]
Thanks, omichel.
